modular_skyrat/master_files/code/modules/mob/living/simple_animal/friendly/dogs.dm

```diff
@@ -0,0 +1,169 @@
+			var/datum/ai_controller/dog/EN = src.ai_controller
+			if(!(WEAKREF(target) in EN.blackboard[BB_DOG_FRIENDS]))
+				shootAt(target)
+			if(src.health < 15)
+				EN.current_movement_target = target
+				EN.blackboard[BB_DOG_HARASS_TARGET] = WEAKREF(target)
+				EN.current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/harass)
+
+/mob/living/simple_animal/pet/dog/corgi/borgi/bullet_act(obj/projectile/proj)
+	if(istype(proj, /obj/projectile/beam) || istype(proj, /obj/projectile/bullet))
+		var/mob/living/carbon/human/target = proj.firer
+		if(isliving(target) && src.health > 0)
+			if(!proj.nodamage && proj.damage > 10)
+				shootAt(target)
+				var/datum/ai_controller/dog/EN = src.ai_controller
+				EN.current_movement_target = target
+				EN.blackboard[BB_DOG_HARASS_TARGET] = WEAKREF(target)
+				EN.current_behaviors += GET_AI_BEHAVIOR(/datum/ai_behavior/harass)
+			else
+				shootToyAt(target)
+	return BULLET_ACT_HIT
+
+/mob/living/simple_animal/pet/dog/corgi/borgi/emag_act(user as mob)
+	if(!emagged)
+		emagged = 1
+		visible_message(span_warning("[user] swipes a card through [src].</span>"),span_notice("You overload [src]s internal reactor.</span>"))
+		addtimer(CALLBACK(src, .proc/explode), 60 SECONDS)
+
+/mob/living/simple_animal/pet/dog/corgi/borgi/proc/explode()
+	visible_message(span_warning("[src] makes an odd whining noise.</span>"))
+	explosion(get_turf(src), 0, 1, 6, 9, 2, TRUE)
+	death()
+
+/mob/living/simple_animal/pet/dog/corgi/borgi/proc/shootAt(atom/movable/target)
+	var/turf/T = get_turf(src)
+	var/turf/U = get_turf(target)
+	if(!T || !U)
+		return
+	var/obj/projectile/beam/laser = new /obj/projectile/beam(loc)
+	laser.icon = 'icons/effects/genetics.dmi'
+	laser.icon_state = "eyelasers"
+	playsound(src.loc, 'sound/weapons/taser.ogg', 75, 1)
+	laser.preparePixelProjectile(target, T)
+	laser.firer = src
+	laser.fired_from = src
+	laser.fire()
+
+/mob/living/simple_animal/pet/dog/corgi/borgi/proc/shootToyAt(atom/movable/target)
+	var/turf/T = get_turf(src)
+	var/turf/U = get_turf(target)
+	if(!T || !U)
+		return
+	var/obj/projectile/bullet/reusable/foam_dart/FD = new /obj/projectile/bullet/reusable/foam_dart(loc)
+	FD.icon = 'icons/obj/guns/toy.dmi'
+	FD.icon_state = "foamdart_proj"
+	playsound(src.loc, 'sound/items/syringeproj.ogg', 75, 1)
+	FD.preparePixelProjectile(target, T)
+	FD.firer = src
+	FD.fired_from = src
+	FD.fire()
+
+/mob/living/simple_animal/pet/dog/corgi/borgi/Life(seconds, times_fired)
+	..()
+	//spark for no reason
+	if(prob(5))
+		do_sparks(3, 1, src)
+
+/mob/living/simple_animal/pet/dog/corgi/borgi/handle_automated_action()
+	if(emagged && prob(25))
+		var/mob/living/carbon/target = locate() in view(10, src)
+		if(target)
+			shootAt(target)
+
+/mob/living/simple_animal/pet/dog/corgi/borgi/death(gibbed)
+	// Only execute the below if we successfully died
+	. = ..(gibbed)
+	if(!.)
+		return FALSE
+	do_sparks(3, 1, src)
+	var/datum/ai_controller/dog/EN = src.ai_controller
```

Comment:
Ditto. Src is useful when you're passing something into a proc. Otherwise this is pointless.